tests/Backend/Snowflake/TypedTableInWorkspaceTest.php

```diff
@@ -109,7 +109,249 @@ public function testUnloadFromWSToTypedTable(): void
         $this->unloadAndAssert($workspace['id'], $tableId);
     }
 
-    public function testUnloadFromWSToTypedTableCTAS(): void
+    public function testUnloadFromWSToTypedTableWithoutDeduplication(): void
+    {
+        // create workspace and source table in workspace
+        $workspace = $this->initTestWorkspace();
+
+        $tableId = 'Languages3';
+
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+        $backend->dropTableIfExists($tableId);
+
+        $connection = $workspace['connection'];
+
+        $db = $backend->getDb();
+
+        $quotedTableId = sprintf(
+            '"%s"."%s"',
+            $connection['schema'],
+            $tableId,
+        );
+        // exact length is required
+        $sql = sprintf(
+            '
+            CREATE TABLE %s (
+                "id" INT NOT NULL,
+                "name" VARCHAR(16777216),
+                PRIMARY KEY ("id")
+            )
+        ',
+            $quotedTableId,
+        );
+        $db->query($sql);
+        $db->query(sprintf("INSERT INTO %s VALUES (1, 'john');", $quotedTableId));
+        $db->query(sprintf("INSERT INTO %s VALUES (2, 'does');", $quotedTableId));
+
+        $this->unloadAndAssert($workspace['id'], $tableId);
+
+        // test unload from workspace with _timestamp column exist
+        $db->query(sprintf('DROP TABLE %s', $quotedTableId));
+        $sql = sprintf(
+            '
+            CREATE TABLE %s (
+                "id" INT NOT NULL,
+                "name" VARCHAR(16777216),
+                PRIMARY KEY ("id")
+            )
+        ',
+            $quotedTableId,
+        );
+        $db->query($sql);
+        $db->query(sprintf("INSERT INTO %s VALUES (1, 'john');", $quotedTableId));
+        $db->query(sprintf("INSERT INTO %s VALUES (2, 'does');", $quotedTableId));
+        // snowflake does not enforce primary key so this is allowed
+        // normally such value would be deduplicated in storage
+        // but ctas-om does not deduplicate values
+        $db->query(sprintf("INSERT INTO %s VALUES (2, 'doesToo');", $quotedTableId));
+
+        $runId = $this->_client->generateRunId();
+        $this->_client->setRunId($runId);
+        $this->initEvents($this->_client);
+
+        // do full load
+        $this->_client->writeTableAsyncDirect($this->tableId, [
+            'dataWorkspaceId' => $workspace['id'],
+            'dataTableName' => $tableId,
+            'deduplicationStrategy' => DeduplicationStrategy::INSERT->value,
```

Comment:
Mělo by to jet takto, zkus.
```suggestion
            'deduplicationStrategy' => DeduplicationStrategy::INSERT,
```

Reply:
Ok, ok. Našel jsem, kde se dělá ten `json_decode` https://github.com/guzzle/guzzle/blob/7.9/src/Client.php#L377.

Reply:
Upraveno, otestováno, že funguje.